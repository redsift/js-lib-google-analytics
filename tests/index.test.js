/* globals test, expect, jest */
const {
  default: setupGoogleAnalytics,
  getDefaultProjectSetup,
  projectNames,
  gaAll,
} = require('../dist/js-lib-google-analytics.umd.min');

// const setupGoogleAnalytics = require('../src/index').default;

test('should throw error for missing config', () => {
  global.ga = jest.fn();

  expect(() => setupGoogleAnalytics()).toThrow(
    'Please provide a project configuration!'
  );
});

test('should throw error for missing `uaProjectId` in config', () => {
  global.ga = jest.fn();

  const config = getDefaultProjectSetup();

  expect(() => setupGoogleAnalytics(config)).toThrow(
    'Please provide a "uaProjectId"!'
  );
});

test('should setup a single google UA project (no autolink)', () => {
  global.ga = jest.fn();

  const config = {
    uaProjectId: '_UA_PROJECT_ID_',
    ...getDefaultProjectSetup(),
  };

  setupGoogleAnalytics(config);

  expect(ga.mock.calls).toMatchSnapshot();
});

test('should setup a single google UA project (with autolink)', () => {
  global.ga = jest.fn();

  const config = {
    uaProjectId: '_UA_PROJECT_ID_',
    ...getDefaultProjectSetup(),
    autoLink: ['www.domain1.at', 'www.domain2.at'],
  };

  setupGoogleAnalytics(config);

  expect(ga.mock.calls).toMatchSnapshot();
});

test('should setup two google UA project (with autolink)', () => {
  global.ga = jest.fn();

  const configs = [
    {
      uaProjectId: '_UA_PROJECT_ID_0_',
      ...getDefaultProjectSetup(),
      autoLink: ['www.domain1.at', 'www.domain2.at'],
    },
    {
      uaProjectId: '_UA_PROJECT_ID_1_',
      ...getDefaultProjectSetup(),
      autoLink: ['www.domain3.at', 'www.domain3.at'],
    },
  ];

  setupGoogleAnalytics(configs);

  expect(ga.mock.calls).toMatchSnapshot();
});

test('should setup a single UA project (without autotrack.js)', () => {
  global.ga = jest.fn();

  const config = {
    uaProjectId: '_UA_PROJECT_ID_',
    ...getDefaultProjectSetup(),
    autoTrack: null,
  };

  setupGoogleAnalytics(config);

  expect(ga.mock.calls).toMatchSnapshot();
});

test('should send an event for two configured projects', () => {
  global.ga = jest.fn();

  const configs = [
    {
      uaProjectId: '_UA_PROJECT_ID_0_',
      ...getDefaultProjectSetup(),
    },
    {
      uaProjectId: '_UA_PROJECT_ID_1_',
      ...getDefaultProjectSetup(),
    },
  ];

  setupGoogleAnalytics(configs);

  gaAll('send', 'pageview');

  expect(ga.mock.calls).toMatchSnapshot();
});

// test('should output two project names', () => {
//   global.ga = jest.fn();

//   const configs = [
//     {
//       uaProjectId: '_UA_PROJECT_ID_0_',
//       ...getDefaultProjectSetup(),
//     },
//     {
//       uaProjectId: '_UA_PROJECT_ID_1_',
//       ...getDefaultProjectSetup(),
//     },
//   ];

//   setupGoogleAnalytics(configs);

//   expect(projectNames).toMatchSnapshot();
// });