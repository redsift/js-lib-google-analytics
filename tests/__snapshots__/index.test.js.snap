// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should send an event for two configured projects 1`] = `
Array [
  Array [
    "create",
    "_UA_PROJECT_ID_0_",
    "auto",
    Object {
      "allowLinker": false,
      "name": "_UA_PROJECT_ID_0_",
    },
  ],
  Array [
    "_UA_PROJECT_ID_0_.set",
    "anonymizeIp",
    true,
  ],
  Array [
    "_UA_PROJECT_ID_0_.require",
    "cleanUrlTracker",
    Object {
      "indexFilename": "index.html",
      "queryDimensionIndex": 1,
      "stripQuery": true,
      "trailingSlash": "remove",
    },
  ],
  Array [
    "_UA_PROJECT_ID_0_.require",
    "urlChangeTracker",
  ],
  Array [
    "_UA_PROJECT_ID_0_.send",
    "pageview",
  ],
  Array [
    "create",
    "_UA_PROJECT_ID_1_",
    "auto",
    Object {
      "allowLinker": false,
      "name": "_UA_PROJECT_ID_1_",
    },
  ],
  Array [
    "_UA_PROJECT_ID_1_.set",
    "anonymizeIp",
    true,
  ],
  Array [
    "_UA_PROJECT_ID_1_.require",
    "cleanUrlTracker",
    Object {
      "indexFilename": "index.html",
      "queryDimensionIndex": 1,
      "stripQuery": true,
      "trailingSlash": "remove",
    },
  ],
  Array [
    "_UA_PROJECT_ID_1_.require",
    "urlChangeTracker",
  ],
  Array [
    "_UA_PROJECT_ID_1_.send",
    "pageview",
  ],
  Array [
    "_UA_PROJECT_ID_0_.send",
    "pageview",
    undefined,
  ],
  Array [
    "_UA_PROJECT_ID_1_.send",
    "pageview",
    undefined,
  ],
]
`;

exports[`should setup a single UA project (without autotrack.js) 1`] = `
Array [
  Array [
    "create",
    "_UA_PROJECT_ID_",
    "auto",
    Object {
      "allowLinker": false,
      "name": "_UA_PROJECT_ID_",
    },
  ],
  Array [
    "_UA_PROJECT_ID_.set",
    "anonymizeIp",
    true,
  ],
  Array [
    "_UA_PROJECT_ID_.send",
    "pageview",
  ],
]
`;

exports[`should setup a single google UA project (no autolink) 1`] = `
Array [
  Array [
    "create",
    "_UA_PROJECT_ID_",
    "auto",
    Object {
      "allowLinker": false,
      "name": "_UA_PROJECT_ID_",
    },
  ],
  Array [
    "_UA_PROJECT_ID_.set",
    "anonymizeIp",
    true,
  ],
  Array [
    "_UA_PROJECT_ID_.require",
    "cleanUrlTracker",
    Object {
      "indexFilename": "index.html",
      "queryDimensionIndex": 1,
      "stripQuery": true,
      "trailingSlash": "remove",
    },
  ],
  Array [
    "_UA_PROJECT_ID_.require",
    "urlChangeTracker",
  ],
  Array [
    "_UA_PROJECT_ID_.send",
    "pageview",
  ],
]
`;

exports[`should setup a single google UA project (with autolink) 1`] = `
Array [
  Array [
    "create",
    "_UA_PROJECT_ID_",
    "auto",
    Object {
      "allowLinker": true,
      "name": "_UA_PROJECT_ID_",
    },
  ],
  Array [
    "_UA_PROJECT_ID_.set",
    "anonymizeIp",
    true,
  ],
  Array [
    "_UA_PROJECT_ID_.require",
    "linker",
  ],
  Array [
    "_UA_PROJECT_ID_.linker:autoLink",
    Array [
      "www.domain1.at",
      "www.domain2.at",
    ],
  ],
  Array [
    "_UA_PROJECT_ID_.require",
    "cleanUrlTracker",
    Object {
      "indexFilename": "index.html",
      "queryDimensionIndex": 1,
      "stripQuery": true,
      "trailingSlash": "remove",
    },
  ],
  Array [
    "_UA_PROJECT_ID_.require",
    "urlChangeTracker",
  ],
  Array [
    "_UA_PROJECT_ID_.send",
    "pageview",
  ],
]
`;

exports[`should setup two google UA project (with autolink) 1`] = `
Array [
  Array [
    "create",
    "_UA_PROJECT_ID_0_",
    "auto",
    Object {
      "allowLinker": true,
      "name": "_UA_PROJECT_ID_0_",
    },
  ],
  Array [
    "_UA_PROJECT_ID_0_.set",
    "anonymizeIp",
    true,
  ],
  Array [
    "_UA_PROJECT_ID_0_.require",
    "linker",
  ],
  Array [
    "_UA_PROJECT_ID_0_.linker:autoLink",
    Array [
      "www.domain1.at",
      "www.domain2.at",
    ],
  ],
  Array [
    "_UA_PROJECT_ID_0_.require",
    "cleanUrlTracker",
    Object {
      "indexFilename": "index.html",
      "queryDimensionIndex": 1,
      "stripQuery": true,
      "trailingSlash": "remove",
    },
  ],
  Array [
    "_UA_PROJECT_ID_0_.require",
    "urlChangeTracker",
  ],
  Array [
    "_UA_PROJECT_ID_0_.send",
    "pageview",
  ],
  Array [
    "create",
    "_UA_PROJECT_ID_1_",
    "auto",
    Object {
      "allowLinker": true,
      "name": "_UA_PROJECT_ID_1_",
    },
  ],
  Array [
    "_UA_PROJECT_ID_1_.set",
    "anonymizeIp",
    true,
  ],
  Array [
    "_UA_PROJECT_ID_1_.require",
    "linker",
  ],
  Array [
    "_UA_PROJECT_ID_1_.linker:autoLink",
    Array [
      "www.domain3.at",
      "www.domain3.at",
    ],
  ],
  Array [
    "_UA_PROJECT_ID_1_.require",
    "cleanUrlTracker",
    Object {
      "indexFilename": "index.html",
      "queryDimensionIndex": 1,
      "stripQuery": true,
      "trailingSlash": "remove",
    },
  ],
  Array [
    "_UA_PROJECT_ID_1_.require",
    "urlChangeTracker",
  ],
  Array [
    "_UA_PROJECT_ID_1_.send",
    "pageview",
  ],
]
`;
